addons/hr_timesheet/views/project_sharing_views.xml

```diff
@@ -146,12 +146,12 @@
             <field name="portal_user_names" position="after">
                 <field name="allow_subtasks" invisible="1"/>
                 <field name="allow_timesheets" invisible="1"/>
-                <field name="planned_hours" widget="timesheet_uom_no_toggle" sum="Initially Planned Hours" optional="hide" attrs="{'column_invisible': [('allow_timesheets', '=', False)]}"/>
-                <field name="effective_hours" widget="timesheet_uom" sum="Effective Hours" optional="show" attrs="{'column_invisible': [('allow_timesheets', '=', False)]}"/>
-                <field name="remaining_hours" widget="timesheet_uom" sum="Remaining Hours" optional="hide" decoration-danger="progress &gt;= 100" decoration-warning="progress &gt;= 80 and progress &lt; 100" attrs="{'column_invisible': [('allow_timesheets', '=', False)]}"/>
-                <field name="subtask_effective_hours" widget="timesheet_uom" attrs="{'column_invisible' : ['|', ('allow_subtasks', '=', False), ('allow_timesheets', '=', False)]}" optional="hide"/>
-                <field name="total_hours_spent" widget="timesheet_uom" attrs="{'column_invisible' : ['|', ('allow_subtasks', '=', False), ('allow_timesheets', '=', False)]}" optional="hide"/>
-                <field name="progress" widget="progressbar" attrs="{'column_invisible': [('allow_timesheets', '=', False)]}" optional="show"/>
+                <field name="planned_hours" widget="timesheet_uom_no_toggle" sum="Initially Planned Hours" optional="hide" attrs="{'invisible': [('allow_timesheets', '=', False)]}"/>
```

Comment:
It is not really `an invalid use of the "colunm_invisible" without parent view`. In fact, it is to be able to hide the column when `allow_timesheets=False` by supposing the fields used in the domain in `column_invisible` is a related field pointing a field in project.

(I have made a custo JS to be able to use `column_invisible` without any parent).

Here with your changes, if `allow_timesheets=False` you will just keep the column and hide the cell in all rows because the `allow_timesheets` will be False for all tasks since we share only one project in Project Sharing. 🙂

You can check in 16.0, it works pretty well. 🙂